### PR TITLE
feat: désactivations des users jamais connectés et créés depuis + de 3 mois et ceux connectés depuis + de 3 mois, possibilité de réactiver côté superadmin et ajout des compteurs superadmin

### PR DESCRIPTION
**Pull request by rap2hpoutre**
Je pense que c'est bien si tu relis et teste pour celle là

## Blabla de copilote pour voir
This pull request introduces a feature to manage user accounts by adding functionality to disable and reactivate users, particularly those inactive for over three months. The changes span multiple files, including controllers, models, migrations, and front-end components.

### Backend Changes:
* **Controllers**:
  * [`api/src/controllers/organisation.js`](diffhunk://#diff-4a0d1b2b25911df4dae7b7b646e92749100ecb8e5316df88e8a2f877eb9945faR246): Added logic to fetch disabled users and include their count in the response. [[1]](diffhunk://#diff-4a0d1b2b25911df4dae7b7b646e92749100ecb8e5316df88e8a2f877eb9945faR246) [[2]](diffhunk://#diff-4a0d1b2b25911df4dae7b7b646e92749100ecb8e5316df88e8a2f877eb9945faR289)
  * [`api/src/controllers/user.js`](diffhunk://#diff-c9b7b24b5b498758d92cb175acabd8d41de2c28b488cbdceff2f2bae697f3b18R197-R204): Added checks to prevent disabled users from logging in and added a new endpoint to reactivate users. [[1]](diffhunk://#diff-c9b7b24b5b498758d92cb175acabd8d41de2c28b488cbdceff2f2bae697f3b18R197-R204) [[2]](diffhunk://#diff-c9b7b24b5b498758d92cb175acabd8d41de2c28b488cbdceff2f2bae697f3b18R312-R315) [[3]](diffhunk://#diff-c9b7b24b5b498758d92cb175acabd8d41de2c28b488cbdceff2f2bae697f3b18R1058-R1083)

* **Database Migration**:
  * [`api/src/db/migrations/20241030093635-add-disabled-at-to-users.js`](diffhunk://#diff-2e76b9b411db14c4754d01a48bd390e3a6dda2244eecba686ce74da65e549b56R1-R18): Added a new column `disabledAt` to the `User` table to track when a user is disabled.

* **Models**:
  * [`api/src/models/user.js`](diffhunk://#diff-eeb8aac18d9b9b87371eeadaec200510a5724b79469ad57cc3c7480cb4717a7eR28): Updated the `User` model to include the new `disabledAt` attribute.

* **Utilities**:
  * [`api/src/utils/desactivateUsers.js`](diffhunk://#diff-5396e743bca18d5351df525d2d692c6250442aefbe2c71a1867bfa849ceb5a81R1-R38): Implemented a utility to disable users who have been inactive for over three months.
  * [`api/src/index.js`](diffhunk://#diff-04c9e51516f060e94f808d8ae6ef98ad296bdfeb08387199de5b48e5ec968a1aR18): Included the new utility in the project.

### Frontend Changes:
* **Components**:
  * [`dashboard/src/scenes/organisation/SuperadminOrgationsationUsers.tsx`](diffhunk://#diff-d47780ba19cc95bcd80174af6d835c3c7fe3564892de9ecc7faf55097a6824baR34): Added UI elements to display disabled users and allow superadmins to reactivate them. [[1]](diffhunk://#diff-d47780ba19cc95bcd80174af6d835c3c7fe3564892de9ecc7faf55097a6824baR34) [[2]](diffhunk://#diff-d47780ba19cc95bcd80174af6d835c3c7fe3564892de9ecc7faf55097a6824baR76) [[3]](diffhunk://#diff-d47780ba19cc95bcd80174af6d835c3c7fe3564892de9ecc7faf55097a6824baR127-R155)
  * [`dashboard/src/scenes/organisation/superadmin.jsx`](diffhunk://#diff-61f1038ab1394f2a547deb47f95e488cd91c26323ac417c9d2db0ded4622b177R229): Updated the superadmin view to show the count of disabled users.

* **Types**:
  * [`dashboard/src/types/user.ts`](diffhunk://#diff-ffe782518e438035cd9c18abcb683343e347747a17048b8961c311b26cbbc348R22): Updated the `UserInstance` type to include the `disabledAt` attribute.

### CODE REVIEW

**Comment by arnaudambro** [comment]
le message d'erreur est `Votre compte a été désactivé, veuillez contacter l'administrateur de votre organisation` mais seul un superadmin peut le réactiver

est-ce une erreur ?

File: api/src/controllers/user.js [hunk 1]
```diff
@@ -1044,4 +1057,30 @@ router.post(
...
+      const error = new Error(`Invalid request in reactivate user: ${e}`);
+      error.status = 400;
+      return next(error);
+    }
+
+    const _id = req.body._id;
+    const user = await User.findOne({ where: { _id } });
+    if (!user) return res.status(404).send({ ok: false, error: "Not Found" });
+
+    user.disabledAt = null;
```
Comment (arnaudambro) [inline api/src/controllers/user.js:1079]: si entre le moment où un utilisateur est réactivé et cet utilisateur vient se connecter sur mano, on redémarre l'api, alors cet utilisateur va être désactivé à nouveau via le script `desactivateInactiveUsers`

je ne sais pas si c'est grave ou pas

suggestion: quand on réactive, on pourrait mettre `lastLoginAt: new Date()` ?

Comment (rap2hpoutre) [reply api/src/controllers/user.js:1084]: Bien vu je l'ai rajouté !

**Review by arnaudambro — APPROVED** [review]

**Pull request merged**